src/plenigo/models/ProductBase.php

```diff
@@ -78,8 +84,7 @@ class ProductBase extends ProductId
      * @param float      $prodPrice The product price.
      * @param string     $curr      The currency.
      */
-    public function __construct($id, $prodTitle=null, $prodPrice=null, $curr=null)
-    {
+    public function __construct($id, $prodTitle = null, $prodPrice = null, $curr = null) {
```

Comment:
Failed payment is standalone. So there should be a constructor that only accepts the failed payment flag.